src/command_modules/azure-cli-network/azure/cli/command_modules/network/_params.py

```diff
@@ -295,6 +303,40 @@ def completer(prefix, action, parsed_args, **kwargs):  # pylint: disable=unused-
 register_cli_argument('network application-gateway waf-config', 'disabled_rules', nargs='+')
 register_cli_argument('network application-gateway waf-config list-rule-sets', '_type', options_list=['--type'])
 
+with VersionConstraint(ResourceType.MGMT_NETWORK, min_api='2017-06-01') as c:
+    c.register_cli_argument('network application-gateway ssl-policy predefined', 'predefined_policy_name', name_arg_type)
+
+    c.register_cli_argument('network application-gateway redirect-config', 'redirect_type', options_list=['--type', '-t'], help='HTTP redirection type', **model_choice_list(ResourceType.MGMT_NETWORK, 'ApplicationGatewayRedirectType'))
+    c.register_cli_argument('network application-gateway redirect-config', 'include_path', **three_state_flag())
```

Comment:
For include_path and include_query_string, please note that these are optional parameters and default is true. It would be better if the user supplied true or false values (especially false) got assigned to the redirect object.